internal/service/elbv2/hosted_zone_id_data_source_test.go

```diff
@@ -0,0 +1,68 @@
+package elbv2_test
+
+import (
+	"testing"
+
+	"github.com/aws/aws-sdk-go/service/elbv2"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-provider-aws/internal/acctest"
+	tfelbv2 "github.com/hashicorp/terraform-provider-aws/internal/service/elbv2"
+)
+
+func TestAccLBHostedZoneIDDataSource_basic(t *testing.T) {
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { acctest.PreCheck(t) },
+		ErrorCheck:        acctest.ErrorCheck(t, elbv2.EndpointsID),
+		ProviderFactories: acctest.ProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckAWSLbHostedZoneIdConfig,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("data.aws_lb_hosted_zone_id.main", "id", tfelbv2.HostedZoneIdPerRegionMap[acctest.Region()]),
+				),
+			},
+			{
+				Config: testAccCheckAWSLbHostedZoneIdExplicitRegionConfig,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("data.aws_lb_hosted_zone_id.regional", "id", "Z32O12XQLNTSW2"),
+				),
+			},
+			{
+				Config: testAccCheckAWSLbHostedZoneIdExplicitNetworkConfig,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("data.aws_lb_hosted_zone_id.network", "id", tfelbv2.HostedZoneIdPerRegionNLBMap[acctest.Region()]),
+				),
+			},
+			{
+				Config: testAccCheckAWSLbHostedZoneIdExplicitNetworkRegionConfig,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("data.aws_lb_hosted_zone_id.network-regional", "id", "Z2IFOLAFXWLO4F"),
+				),
+			},
+		},
+	})
+}
+
+const testAccCheckAWSLbHostedZoneIdConfig = `
+data "aws_lb_hosted_zone_id" "main" {
+}
```

Comment:
Since there are no arguments passed for this test we can refactor it to the following:

```suggestion
data "aws_lb_hosted_zone_id" "main" {}

Reply:
make sense, added in [82569e5](https://github.com/hashicorp/terraform-provider-aws/pull/24749/commits/82569e539f4015104fa62478b329eb38cf4e7f40)